govcd/catalog_test.go

```diff
@@ -207,6 +207,55 @@ func doesCatalogExist(check *C, org *AdminOrg) {
 	check.Assert(err, NotNil)
 }
 
+// Creates a Catalog, uploads a vApp template to it, renames it, retrieves it
+// using the updated name and checks if it has the same vApp template.
+// If it doesn't the assertion fails.
+func (vcd *TestVCD) Test_RenameCatalog(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	org, err := vcd.client.GetAdminOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	check.Assert(org, NotNil)
+
+	adminCatalog, err := org.CreateCatalog(TestRenameCatalog, TestRenameCatalog)
+	check.Assert(err, IsNil)
+	check.Assert(adminCatalog, NotNil)
+
+	task, err := adminCatalog.UploadOvf(vcd.config.OVA.OvaPath, TestUploadOvf, TestUploadOvf, 1024)
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	catalog, err := vcd.client.Client.GetCatalogByName(vcd.config.VCD.Org, TestRenameCatalog)
+	check.Assert(err, IsNil)
+	check.Assert(catalog, NotNil)
+
+	vAppTemplate1, err := catalog.GetVAppTemplateByName(TestUploadOvf)
+	check.Assert(err, IsNil)
+	check.Assert(vAppTemplate1, NotNil)
+
+	adminCatalog.AdminCatalog.Name = TestRenameCatalog + "_updated"
+	err = adminCatalog.Update()
+	check.Assert(err, IsNil)
+
+	// The catalog is added to the cleanup list only after the name update, as
+	// the old one can't be found.
+	AddToCleanupList(TestRenameCatalog+"_updated", "catalog", vcd.config.VCD.Org, check.TestName())
```

Comment:
You should add to the cleanup list immediately after the creation, or else, if something fails between creation and update, the resource will not be removed. 
You can keep this second addition. The cleanup function behaves nicely when the resource was already removed.

Reply:
Fixed.